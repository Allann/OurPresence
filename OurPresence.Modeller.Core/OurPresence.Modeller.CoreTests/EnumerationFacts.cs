using OurPresence.Modeller.Domain;
using OurPresence.Modeller.Domain.Extensions;
using Xunit;
using FluentAssertions;
using ApprovalTests.Reporters;
using ApprovalTests;

namespace OurPresence.Modeller.CoreTests
{
    [UseReporter(typeof(DiffReporter))]
    public class EnumerationFacts
    {
        [Fact]
        public void Enumeration_CanSerialize_Success()
        {
            var module = new Module("Company", "Test");
            var sut = new Enumeration("PaymentTypes")
                .AddItem("Cash")
                .AddItem("CreditCard")
                .AddItem("Cheque")
                .AddItem("DirectDebit");
            module.Enumerations.Add(sut);

            var expected = module.ToJson();

            Approvals.VerifyJson(expected);
            
            var actual = expected.FromJson<Module>();
            actual.Should().BeEquivalentTo(module);
        }

        [Fact]
        public void Enumeration_FlagValues_Set()
        {
            var module = new Module("Company", "Test");
            var sut = new Enumeration("PaymentTypes", true)
                .AddItem("None")
                .AddItem("Cash")
                .AddItem("CreditCard")
                .AddItem("Cheque")
                .AddItem("DirectDebit");
            module.Enumerations.Add(sut);

            var expected = module.ToJson();

            Approvals.VerifyJson(expected);

            var actual = expected.FromJson<Module>();
            actual.Should().BeEquivalentTo(module);
        }
    }
}
