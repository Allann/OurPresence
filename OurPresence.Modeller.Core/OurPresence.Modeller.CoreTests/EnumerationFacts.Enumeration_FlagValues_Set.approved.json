{
    "company": "Company",
    "project": "Test",
    "models": [],
    "enumerations": [
        {
            "flag": true,
            "items": [
                {
                    "name": "None",
                    "display": "None",
                    "value": 0
                },
                {
                    "name": "Cash",
                    "display": "Cash",
                    "value": 1
                },
                {
                    "name": "CreditCard",
                    "display": "Credit Card",
                    "value": 2
                },
                {
                    "name": "Cheque",
                    "display": "Cheque",
                    "value": 4
                },
                {
                    "name": "DirectDebit",
                    "display": "Direct Debit",
                    "value": 8
                }
            ],
            "name": "PaymentType"
        }
    ],
    "requests": []
}